Infrastructure/Repository.cs

```diff
@@ -19,19 +20,21 @@ public Repository(ApplicationDbContext db, ILogger<Repository<T>> logger)
             _logger = logger;
         }
 
-        public async Task<bool> Add(T entity)
+        public async Task<int> Add(T entity)
         {
+            EntityEntry<T> addedEntity = null;
             try
             {
-                await _dbSet.AddAsync(entity);
+                addedEntity = await _dbSet.AddAsync(entity);
+                await Commit();
             }
             catch (Exception ex)
             {
                 _logger.LogError(ex, $"Repository {nameof(T)} add error");
                 throw;
             }
 
-            return await Commit();
+            return addedEntity.Entity.Id;
```

Comment:
This code is not future-proof anymore.
Please make sure to cover the potential case of adding an Entity with GUID id

Reply:
There are several ways to do it:
1. Add another generic parameter to IRepository indicating the Id type
2. Return a complete added / updated entity, instead of just its id
3. Use separate interfaces for different entities

Reply:
done by 1 and 2 ways